### ISSUE TITLE
documentation: using ALB with DSR

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-azure-arm-templates#73]
## Description

Working with a Microsoft Azure Architect we had a discussion/POC with a customer where MS had created a configuration of using a Public IP address configured with DSR to the BIG-IP in the template "failover-lb".  

This met a requirement of being able to see the Public IP directly on the BIG-IP and not perform any address/port translation.  It does complicate how to create a health probe.  MS had used the MGMT Web UI as a health probe, I've created specific iRules that return "OK".

Ideally we would have guidance on a sane setup for supporting this configuration or caveats about using this setup.  IMHO it's a good solution.

## Template

"cluster-lb"

## Severity Level
For bugs, enter the bug severity level. Do not set any labels.

Severity:  5

**Comment by jmcalalang** [comment]
I've done this at a customer site, actually with the recommendation of a MS SA as well... one thing i ran into was this worked for single BIG-IP units, but wouldnt for a cluster, if needed i can test again why it didnt work (it was almost a year ago).

the SA recommended this to get around a limitation of the of the azure lb at the time with port exhaustion problems.

It shows and it pretty fantastic, health monitoring is a weird piece though

![unknown](https://user-images.githubusercontent.com/20847865/39227240-6fdd9322-480c-11e8-91ec-364bc7f083f8.png)
![unknown-1](https://user-images.githubusercontent.com/20847865/39227241-6ff1f1e6-480c-11e8-83c4-24f2ff32da9f.png)
![unknown-2](https://user-images.githubusercontent.com/20847865/39227242-70047a46-480c-11e8-883c-57dc705c9c89.png)
![unknown-3](https://user-images.githubusercontent.com/20847865/39227243-702b2c04-480c-11e8-88cb-0be6659bd8cb.png)
![unknown-4](https://user-images.githubusercontent.com/20847865/39227244-70401236-480c-11e8-9572-69c3e48644fb.png)
![unknown-5](https://user-images.githubusercontent.com/20847865/39227245-705416aa-480c-11e8-9320-29313682c6a1.png)

**Comment by jmcalalang** [comment]
I did this for a customer about a year ago on a recommendation of a MS SA for port exhaustion on an Azure LB, it works like a champ, of course it required the LB out front so it took out the HA-AVSET option from the customer i was working with.

Name removed for the VIP, if you need the actual customer let me know.

![unknown](https://user-images.githubusercontent.com/20847865/39227384-2b776716-480d-11e8-87d9-9519cd78d1cc.png)
![unknown-1](https://user-images.githubusercontent.com/20847865/39227385-2b9e3fee-480d-11e8-8c7d-df4682b198be.png)
![unknown-2](https://user-images.githubusercontent.com/20847865/39227386-2bc47bd2-480d-11e8-8cc0-723784c0d132.png)
![unknown-3](https://user-images.githubusercontent.com/20847865/39227387-2bd88b7c-480d-11e8-9526-c867b8d25e87.png)
![unknown-4](https://user-images.githubusercontent.com/20847865/39227388-2becdfaa-480d-11e8-86b2-bb7b6ccb5151.png)
![unknown-5](https://user-images.githubusercontent.com/20847865/39227389-2c014d50-480d-11e8-998d-6023fb5cdc3e.png)

**Comment by chen23** [comment]
FYI, here's a (lame) iRule that I used to send back a happy health check (for A/A):

**A/A**

```
when HTTP_REQUEST {
  HTTP::respond 200 content "OK"
}
```
Apply this to the IP:PORT that ALB is configured to monitor.  i.e. 10.0.1.5:8888 on bigip1 and 10.0.1.6 on bigip2 these could either be the selfip (primary private address) or secondary.  For some reason I think ALB prefers to monitor the primary (tcpdump is your friend to verify).

This should only respond when BIG-IP is fully up (management ports could be up before the device is able to pass traffic).

**A/S**

The other option for a A/S (there may be a better way?):

Use the above iRule on a virtual that is not on routable space (255.255.255.254:8888) and assign it to the preferred traffic-group (i.e. traffic-group-1).

Add the following iRule to the IP:PORT that ALB uses:
```
when CLIENT_ACCEPTED {
    virtual float_is_alive_vs
}
```
Assuming that the "real" monitor is on a virtual named "float_is_alive_vs" this will only be able to respond to ALB if the device is "active".  This is following the behavior that is documented at: https://support.f5.com/csp/article/K10379 where you can use the virtual command to route traffic from one virtual to the another.  The first virtual is set for traffic-group-none and will always respond to traffic.  The second virtual will only respond to traffic if the device is active for that particular traffic-group.

A potential optimization would be to reference the actual virtual that you want to monitor (you would need to define a separate ALB health probe for each service similar to when you do not use DSR).

I also think that using "traffic-group-none", on the virtual that handles traffic, keeps things flowing a little bit better b/c the BIG-IP will still accept a flow regardless of what the device thinks is the current "active" device (this trick only works in a case like Azure where the BIG-IP is not able to ARP for IPs, this is a bit of a disaster if you try this in a "normal" network b/c both devices would be trying to ARP and you would get IP conflicts).

**Comment by edwoodjrjr** [comment]
Internal issue 1179.

**Comment by rftoccijr** [comment]
+1 on the DSR.  I used this recently for a customer because he was given carved up networks in /28.  This limited us to 11 IPs in the subnmet itself (16 addreses minus network and broadcase, minus the top three used by Azure as a gateway and dns).  

If I didn't use DSR, then I'd have to spend 2 private IPs in the range (one for each BIG-IP so I can "fail over"), meaning I can only deploy 5 applications.  I could port map, but that makes the configuration even more complicated.

This allowed me to use up to 20 public IPs and just push them to the BIG-IP.

**Comment by shyawnkarim** [comment]
Closing due to age. These legacy templates are now in maintenance mode and are being replaced by our next-generation templates available in the [Cloud Templates 2.0 GitHub repo](https://github.com/F5Networks/f5-azure-arm-templates-v2).
